test/kafka_ex/network_client_test.exs

```diff
@@ -0,0 +1,68 @@
+defmodule KafkaEx.NetworkClientTest do
+  use ExUnit.Case, async: true
+
+  use Hammox.Protect,
```

Comment:
I've not used Hammox, but reading through and doing some testing myself, it seems like this call just creates wrappers that will additionally fail if dialyzer types don't match? In testing locally, the function calls here are definitely still calling through to the originals, so there are no real mocks created.

Reply:
I'm using Hammox quite a lot.
You have two modes there, one with `use Hamox.Protect` it will wrap function & check in in run time same way as dialyzer in compile time. It's quite useful as it allows you to validate behaviors when dialyzer sometimes is too optimistic.